docs/sphinx/cookbook/coalesce_pool.rst

```diff
@@ -4,10 +4,49 @@
 Coalescing Pool Memory
 ======================
 
-The :class:`umpire::strategy::QuickPool` provides a
-:func:`umpire::strategy::QuickPool::coalesce` that can be used to release
+Umpire's memory pools provide a more performant way to allocate large
+amounts of memory with fewer calls to the underlying API. 
+Memory pools allow developers to allocate all needed memory 
+at once instead of making multiple, smaller memory allocations which can 
+become quite expensive. However, the performance of memory pools 
+varies widely depending upon how the memory blocks within the pool are managed.
+Each allocation within a memory pool must fit within a `memory block`.
+As memory is allocated and deallocated, those blocks must be properly 
+adjusted in order to handle new, incoming memory allocations.
+
+Umpire's solution for managing blocks of memory within a pool is to use a
+`coalescing function`. The coalescing function will deallocate a certain amount
+of unused memory and reallocate it, creating a new memory block which can handle new allocations.
+
```

Comment:
I would replace the above paragraph with something like the following:
```suggestion
When Umpire is unable to find a block with sufficient memory for the applications' allocation
request, it will allocate a new block from the allocation resource and add that block to the pool.
As application memory use grows, the pool will fill up with many blocks.  Umpire has a `coalescing function` that will deallocate all of the unused (free) blocks in its pool and then reallocate a single block of memory that is large enough to contain all of the freed blocks.
```